runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,901 @@
+        if len(args) > 0 and self.args_names is not None:
+            payload_invoke = {self.args_names[i]: args[i] for i in range(len(args))}
+        invoke_res = self.lambda_client.invoke(
+            FunctionName=self.name,
+            Payload=json.dumps({**payload_invoke, **kwargs}),
+            LogType="Tail",
+        )
+        return_value = invoke_res["Payload"].read().decode("utf-8")
+
+        try:
+            logger.error(invoke_res["FunctionError"])
+            return_value = json.loads(return_value)
+            raise RuntimeError(
+                f"Failed to run {self.name}. Error: {return_value['errorType']}, {return_value['errorMessage']}"
+            )
+
+        except KeyError:
+            if invoke_for_the_first_time:
+                lambdas_log_groups = self._get_log_group_names()
+                with self._wait_till_log_group_is_created(
+                    curr_lambda_log_group, lambdas_log_groups
+                ):
+                    self.logs_client.put_retention_policy(
+                        logGroupName=curr_lambda_log_group,
+                        retentionInDays=self.retention_time,
+                    )
+
+            log_lines = "Function Logs are:\n" + base64.b64decode(
+                invoke_res["LogResult"]
+            ).decode("utf-8")
+
+            print(log_lines)
+            return return_value
+
+    def map(self, *args, **kwargs):
+        """Map a function over a list of arguments.
+
+        Example:
+            >>> # The my_lambda_handler.py file
+            >>> def my_summer(arg1, arg2, arg3):
+            >>>     return arg1 + arg2 + arg3
+            >>>
+            >>> # your 'main' python file, where you are using runhouse
+            >>> summer_lambda = rh.lambda_function(
+            >>>                     paths_to_code=['/full/path/to/handler_file.py'],
+            >>>                     handler_function_name = 'summer',
+            >>>                     runtime = 'python3.9',
+            >>>                     name="my_func").to()
+            >>> output = summer_lambda.map([1, 2], [1, 4], [2, 3])  # output = ["4", "9"]
+
+
+        """
+
+        return [self._invoke(*args, **kwargs) for args in zip(*args)]
+
+    #
+    def starmap(self, args_lists, **kwargs):
+        """Like :func:`map` except that the elements of the iterable are expected to be iterables
+        that are unpacked as arguments. An iterable of [(1,2), (3, 4)] results in [func(1,2), func(3,4)].
+
+        Example:
+            >>> arg_list = [(1,2, 3), (3, 4, 5)]
+            >>> # invokes the Lambda function twice, once with args (1, 2, 3) and once with args (3, 4, 5)
+            >>> output = summer_lambda.starmap(arg_list) # output = ["6", "12"]
+        """
+
+        return [self._invoke(*args, **kwargs) for args in args_lists]
+
+    # --------------------------------------
+    # Lambda Function delete methods
+    # --------------------------------------
+    def delete(self):
```

Comment:
For clusters we have `teardown` and `teardown_and_delete`, the first to delete the infra and the second to infra +  delete the configs from RNS. wdyt about using a similar concept here?